### PR TITLE
Rewrite transforms v2 e2e example

### PR DESCRIPTION
**Pull request by NicolasHug**
This is a light rewrite despite the seemingly big diff. I just extended it a little to include masks and to be a bit more descriptive / detailed.

### CODE REVIEW

File: gallery/v2_transforms/plot_transforms_v2.py [hunk 1]
```diff
@@ -90,7 +90,7 @@
 
 from torchvision import datapoints  # we'll describe this a bit later, bare with us
 
-bboxes = datapoints.BoundingBoxes(
+boxes = datapoints.BoundingBoxes(
```
Comment (NicolasHug) [inline gallery/v2_transforms/plot_transforms_v2.py:93]: Dataset wrapper uses "boxes" instead of "bboxes". I don't have a pref, but we should align (hence the changes in this example)

File: gallery/v2_transforms/plot_transforms_v2_e2e.py [hunk 2]
```diff
@@ -1,146 +1,169 @@
...
 
 .. note::
     Try on `collab <https://colab.research.google.com/github/pytorch/vision/blob/gh-pages/main/_generated_ipynb_notebooks/plot_transforms_v2_e2e.ipynb>`_
     or :ref:`go to the end <sphx_glr_download_auto_examples_v2_transforms_plot_transforms_v2_e2e.py>` to download the full example code.
 
-Object detection is not supported out of the box by ``torchvision.transforms`` v1, since it only supports images.
-``torchvision.transforms.v2`` enables jointly transforming images, videos, bounding boxes, and masks. This example
-showcases an end-to-end object detection training using the stable ``torchvision.datasets`` and ``torchvision.models``
-as well as the new ``torchvision.transforms.v2`` v2 API.
+Object detection and segmentation tasks are natively supported:
```
Comment (NicolasHug) [inline gallery/v2_transforms/plot_transforms_v2_e2e.py:10]: Using "segmentation" loosely here (i.e. both instance and semantic). It's made more accurate a bit later in the example, the main point being that it doesn't matter which one you're targetting, the transforms work the same.

File: gallery/v2_transforms/plot_transforms_v2.py (hunk 1 above)
Comment (pmeier) [reply gallery/v2_transforms/plot_transforms_v2.py:93]: The reason the dataset wrapper uses `"boxes"` is because our models require that:

https://github.com/pytorch/vision/blob/f7c7bdf5fae5e946e67812c5791d91cec83fccd5/torchvision/models/detection/ssd.py#L333-L341

File: gallery/v2_transforms/plot_transforms_v2_e2e.py [hunk 3]
```diff
@@ -1,146 +1,169 @@
...
     dataset,
     batch_size=2,
-    # We need a custom collation function here, since the object detection models expect a
-    # sequence of images and target dictionaries. The default collation function tries to
-    # `torch.stack` the individual elements, which fails in general for object detection,
-    # because the number of object instances varies between the samples. This is the same for
-    # `torchvision.transforms` v1
+    # We need a custom collation function here, since the object detection
+    # models expect a sequence of images and target dictionaries. The default
+    # collation function tries to :func:`~torch.stack` the individual elements,
```
Comment (NicolasHug) [inline gallery/v2_transforms/plot_transforms_v2_e2e.py:154]: rst formatting is useless here as this is a code comment (will self-address later)

```suggestion
    # collation function tries to torch.stack() the individual elements,
```

File: gallery/v2_transforms/plot_transforms_v2_e2e.py [hunk 4]
```diff
@@ -1,146 +1,168 @@
...
 
-Object detection is not supported out of the box by ``torchvision.transforms`` v1, since it only supports images.
-``torchvision.transforms.v2`` enables jointly transforming images, videos, bounding boxes, and masks. This example
-showcases an end-to-end object detection training using the stable ``torchvision.datasets`` and ``torchvision.models``
-as well as the new ``torchvision.transforms.v2`` v2 API.
+Object detection and segmentation tasks are natively supported:
+``torchvision.transforms.v2`` enables jointly transforming images, videos,
+bounding boxes, and masks.
+
+This example showcases an end-to-end instance segmentation training case using
```
Comment (vfdev-5) [inline gallery/v2_transforms/plot_transforms_v2_e2e.py:14]:
```suggestion
This example showcases an end-to-end object detection / instance segmentation training case using
```

File: gallery/v2_transforms/helpers.py [hunk 5]
```diff
@@ -12,20 +15,30 @@ def plot(imgs):
...
                 img -= img.min()
                 img /= img.max()
 
-            if bboxes is not None:
-                img = draw_bounding_boxes(img, bboxes, colors="yellow", width=3)
+            img = F.to_dtype(img, torch.uint8, scale=True)
+            if boxes is not None:
+                img = draw_bounding_boxes(img, boxes, colors="yellow", width=3)
+            if masks is not None:
+                img = draw_segmentation_masks(img, masks.to(torch.bool), alpha=.65)
```
Comment (pmeier) [inline gallery/v2_transforms/helpers.py:40]: Should we maybe also make the masks yellow or some other color? I fell in the current version the masks is not completely obvious on the image.

**Review by pmeier — APPROVED** [review]
Minor comment left, but LGTM nonetheless. Thanks Nicolas!

File: gallery/v2_transforms/plot_transforms_v2_e2e.py (hunk 4 above)
Comment (NicolasHug) [reply gallery/v2_transforms/plot_transforms_v2_e2e.py:14]: I think it's fine to keep as-is considering the sentence just below. Technically, it *is* an instance segmentation model.

File: gallery/v2_transforms/helpers.py (hunk 5 above)
Comment (NicolasHug) [reply gallery/v2_transforms/helpers.py:40]: hmmm looks like the `color` param is broken, I tried `colors="green"` and `colors=["green"] * masks.shape[0]`, none of which had an effect. Might be worth looking into but I'll merge

Comment (NicolasHug) [reply gallery/v2_transforms/helpers.py]: maybe I was looking at the wrong output - it actually works fine and the masks are green (and very obvious)
